src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -311,7 +325,7 @@ class NdArrayConnectivityField(  # type: ignore[misc] # for __ne__, __eq__
     common.ConnectivityField[common.DimsT, common.DimT],
     NdArrayField[common.DimsT, core_defs.IntegralScalar],
 ):
-    _codomain: common.DimT
+    _codomain: common.DimT | fbuiltins.FieldOffset
```

Comment:
I don't think this is the right direction, the changes related to this, need to be in the `as_offset` implementation.